google/cloud/internal/oauth2_service_account_credentials.cc

```diff
@@ -327,11 +340,17 @@ StatusOr<ServiceAccountCredentialsInfo> ParseServiceAccountP12File(
                                        GoogleOAuthRefreshEndpoint(),
                                        /*scopes=*/{},
                                        /*subject=*/{},
-                                       /*enable_self_signed_jwt=*/false};
+                                       /*enable_self_signed_jwt=*/false,
+                                       /*universe_domain=*/{}};
 }
 #include "google/cloud/internal/diagnostics_pop.inc"
 
 bool ServiceAccountUseOAuth(ServiceAccountCredentialsInfo const& info) {
+  // Custom universe domains are only supported with JWT, not OAuth tokens.
+  if (info.universe_domain.has_value() &&
+      info.universe_domain != GoogleDefaultUniverseDomain()) {
+    return false;
+  }
```

Comment:
FYI: somewhere in the release notes we should highlight that P12 service account key files do not work with universe domains.